nx/test/nx_test.exs

```diff
@@ -2959,4 +2959,65 @@ defmodule NxTest do
       assert_all_close(zeros, Nx.imag(x_ifft), atol: 1.0e-8)
     end
   end
+
+  describe "split/2" do
+    test "Split list into 50% for training and 50% for testing" do
+      tensor = Nx.iota({10, 2}, names: [:x, :y])
+      {train, test} = Nx.split(tensor, split: 0.5)
+
+      assert Nx.tensor(
+               [
+                 [0, 1],
+                 [2, 3],
+                 [4, 5],
+                 [6, 7],
+                 [8, 9]
+               ],
+               names: [:x, :y]
+             ) == train
+
+      assert Nx.tensor(
+               [
+                 [10, 11],
+                 [12, 13],
+                 [14, 15],
+                 [16, 17],
+                 [18, 19]
+               ],
+               names: [:x, :y]
+             ) == test
+    end
+
+    test "Split into 70% for training and 30% for testing" do
+      tensor = Nx.iota({100, 6})
+      {train, test} = Nx.split(tensor, split: 0.7)
+
+      assert length(Nx.to_list(train)) == 70
+      assert length(Nx.to_list(test)) == 30
+    end
+
+    test "Split into 75% for training and 25% for testing" do
+      tensor = Nx.iota({100, 10})
+      {train, test} = Nx.split(tensor, split: 0.75)
+
+      assert length(Nx.to_list(train)) == 75
+      assert length(Nx.to_list(test)) == 25
+    end
+
+    test "Split into 61% for training and 39% for testing" do
+      tensor = Nx.iota({100, 10})
+      {train, test} = Nx.split(tensor, split: 0.61)
+
+      assert length(Nx.to_list(train)) == 61
+      assert length(Nx.to_list(test)) == 39
+    end
+
+    test "Split into 60% for training and 40% for testing with unbalanced data" do
+      tensor = Nx.iota({73, 4})
+      {train, test} = Nx.split(tensor, split: 0.61)
+
+      assert length(Nx.to_list(train)) == 44
+      assert length(Nx.to_list(test)) == 29
+    end
```

Comment:
```suggestion
    test "Split into 60% for training and 40% for testing with unbalanced data" do
      tensor = Nx.iota({99, 4})
      {train, test} = Nx.split(tensor, split: 0.6)

      assert {60, 4} == Nx.shape(train)
      assert {39, 4} == Nx.shape(test)
    end
```

Two-fold review here:

1. (this applies to all other tests) let's assert on the shape instead, to avoid Nx.to_list
2. I think showing what happens with the split being a "nice" value but the length being indivisible by it is better for learning purposes. i.e. I would have no idea what 61% of 73 should be, but it's easy to know that 60% of 100 is 60, and 60% of 99 is either 60 of 59 depending on rounding.